test/e2e/specs/interactivity/router-styles.spec.ts

```diff
@@ -53,6 +53,7 @@ test.describe( 'Router styles', () => {
 
 	test.beforeEach( async ( { page, interactivityUtils: utils } ) => {
 		await page.goto( utils.getLink( 'none' ) );
+		await expect( page.getByTestId( 'hydrated' ) ).toBeVisible();
```

Comment:
My feeling is that the click below sometimes happens before the Interactivity API hydrates the HTML, so at the time it's clicked, it remains unresponsive and nothing happens.

https://github.com/WordPress/gutenberg/blob/293dc0c571b9b1118e6dbd33aca8af488fcd7ab2/test/e2e/specs/interactivity/router-styles.spec.ts#L313

With the check above, all tests should wait until the Interactivity API is correctly hydrated.

Let's see if that fixes the problem. 🙂